src/gt4py/next/program_processors/codegens/gtfn/codegen.py

```diff
@@ -98,15 +98,27 @@ def asfloat(value: str) -> str:
         return value
 
     def visit_Literal(self, node: gtfn_ir.Literal, **kwargs: Any) -> str:
+        # TODO(tehrengruber): isn't this wrong and int32 should be casted to an actual int32?
         match pytype_to_cpptype(node.type):
             case "float":
-                return self.asfloat(node.value) + "f"
+                result = self.asfloat(node.value) + "f"
             case "double":
-                return self.asfloat(node.value)
+                result = self.asfloat(node.value)
             case "bool":
-                return node.value.lower()
+                result = node.value.lower()
             case _:
-                return node.value
+                result = node.value
+        if node.type in ["bool", "int32", "int64", "float32", "float64"]:
```

Comment:
maybe we can just make sure that there is a space in the binary operations? `1 - -1` should be ok

`BinaryExpr = as_fmt("({lhs} {op} {rhs})")`

Reply:
Much nicer. Thanks.